File: civitasRuby/lib/sorpresa_ir_carcel.rb
# encoding:utf-8
module Civitas

  class SorpresaIrCarcel < Sorpresa
  
    def initialize(texto, tablero)
      super(texto)
      @tablero = tablero
    end

    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        super(actual, todos)
        todos.at(actual).encarcelar(@tablero.get_carcel)
      end
    end
    
    def to_string
      return @texto
    end
  
  public_class_method :new
  end
end

File: civitas/src/civitas/SorpresaIrCarcel.java
package civitas;


import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class SorpresaIrCarcel extends Sorpresa {
    
    private Tablero tablero;

    public SorpresaIrCarcel(Tablero tablero, String texto) {
        super(texto);
        this.tablero = tablero;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
    
    @Override
    public String toString() {
        String salida = "Tablero: " + tablero + super.toString();
        return salida;
    }
    
}
